Source/WebKit/Shared/API/APIObject.h

```diff
@@ -261,7 +269,11 @@ class Object
     // Derived classes must override operator new and call newObject().
     void* operator new(size_t) = delete;
 
+#if __has_feature(objc_arc)
+    WeakObjCPtr<id> m_wrapper;
+#else
     NSObject *m_wrapper;
+#endif
```

Comment:
> Does this work when some files are compiled with ARC and other files compiled without it, and they are both manipulating the same objects? I would have expected we’d need to use `WeakObjCPtr` for non-ARC too.

In non-ARC mode (e.g., C++ source), we are basically storing raw pointers.  We should switch both to use `WeakObjCPtr`.  It will simplify the code change as well since you can get rid of the `!__has_feature(objc_arc)` code.

Reply:
I initially tried that and the `WeakObjCPtr` was zeroing out too early in the non-ARC case. I'll dive into that some more and see if I can find a solution.